File: code/python/isHappy.py
class Solution:
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """

        count = 0
        C = []
        
        while n!=0:
            temp = n%10
            count += temp*temp
            n = n//10
            if n == 0:
                if count == 1:
                    return True
                elif count in C:
                    return False
                n = count
                count = 0
                C.append(n)
                

File: code/java/isHappy.java
class Solution {
    public boolean isHappy(int n) {
        int count = 0;
        List<Integer> nums = new ArrayList<Integer>();
        while(true){
            while(n!=0){
               count += (n%10) * (n%10);
                n = n/10;
            }
            if(count == 1)
                return true;
            else
                if(!nums.contains(count)){
                    nums.add(count);
                    n = count;
                    count = 0;
                }
                else
                    return false;
        }
    }
}
